react/components/atoms/input/input.js

```diff
@@ -0,0 +1,100 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, Text, TextInput, View } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles } from "../../../util";
+
+export class Input extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            header: PropTypes.string,
+            placeholder: PropTypes.string,
+            value: PropTypes.string,
+            hasBorder: PropTypes.bool,
+            borderColor: PropTypes.string,
+            onValueUpdate: PropTypes.func
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            placeholder: undefined,
+            borderColor: "#e4e8f0",
+            header: undefined,
+            value: undefined,
+            hasBorder: true,
+            onValueUpdate: undefined
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: this.props.value
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.value === this.props.value) return;
+        this.setState({
+            valueData: this.props.value
+        });
+    }
+
+    onChangeValue = value => {
+        this.setState({ valueData: value }, () => {
+            if (this.props.onValueUpdate) this.props.onValueUpdate(value);
+        });
+    };
+
+    _style = () => {
+        return [
+            styles.input,
+            {
+                borderColor: this.props.hasBorder ? "#e4e8f0" : "transparent",
+                borderBottomWidth: this.props.hasBorder ? 1 : 0,
+                borderTopWidth: this.props.hasBorder ? 1 : 0
+            }
+        ];
+    };
+
+    render() {
+        return (
+            <View style={this._style()} {...this.id("input-ripe")}>
+                <Text style={styles.header}>{this.props.header}</Text>

```

Comment:
perhaps add a check here so that if no header is provided, the text input is centered:
```suggestion
               {Boolean(this.props.header) && (
                    <Text style={styles.header}>{this.props.header}</Text>
                )}

```
| - | - |
| --- | --- |
| Before | After |
| <img width="462" alt="imagem" src="https://user-images.githubusercontent.com/25725586/121879754-170a1780-cd05-11eb-8273-5b38ced35d7e.png"> | <img width="462" alt="imagem" src="https://user-images.githubusercontent.com/25725586/121879856-386b0380-cd05-11eb-805e-0b2a69ecb510.png">|